src/jltypes.c

```diff
@@ -2108,8 +2108,10 @@ void jl_precompute_memoized_dt(jl_datatype_t *dt)
     }
 }
 
+static arraylist_t partial_inst;
```

Comment:
no. I believe executing typedefs in parallel might allow the definition of unsound / undecidable types. Eventually probably should be a BKL around the side-effecting toplevel expressions since they aren't valid in threaded code (including `global`, `using`, `type`, and `function`)